aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md

```diff
@@ -278,7 +284,171 @@ In the preceding example:
+* For demonstration purposes, the <xref:Microsoft.JSInterop.DotNetObjectReference> function call (`invokeMethod` or `invokeMethodAsync`), the .NET method called (`Update` or `UpdateAsync`), and the argument are written to the console. The arguments use a random number to permit matching the JS function call to the .NET method invocation (also written to the console on the .NET side). Production code usually doesn't write to the console, either on the client or the server. Production apps usually rely upon app *logging*. For more information, see <xref:blazor/fundamentals/logging> and <xref:fundamentals/logging/index>.
+
+```html
+<script>
+  const randomInt = () => Math.floor(Math.random() * 99999);
+
+  window.invokeDotNetMethodsAsync = 
+      async (supportSyncInterop, dotNetHelper1, dotNetHelper2) => {
+
+    var n = randomInt();
+    console.log(`JS: invokeMethodAsync:Update('string ${n}')`);
+    await dotNetHelper1.invokeMethodAsync('Update', `string ${n}`);
+
+    n = randomInt();
+    console.log(`JS: invokeMethodAsync:UpdateAsync('string ${n}')`);
+    await dotNetHelper1.invokeMethodAsync('UpdateAsync', `string ${n}`);
+
+    if (supportSyncInterop) {
+      n = randomInt();
+      console.log(`JS: invokeMethod:Update('string ${n}')`);
+      dotNetHelper1.invokeMethod('Update', `string ${n}`);
+    }
+
+    n = randomInt();
+    console.log(`JS: invokeMethodAsync:Update(${n})`);
+    await dotNetHelper2.invokeMethodAsync('Update', n);
+
+    n = randomInt();
+    console.log(`JS: invokeMethodAsync:UpdateAsync(${n})`);
+    await dotNetHelper2.invokeMethodAsync('UpdateAsync', n);
+
+    if (supportSyncInterop) {
+      n = randomInt();
+      console.log(`JS: invokeMethod:Update(${n})`);
+      dotNetHelper2.invokeMethod('Update', n);
+   }
+  };
+</script>
+```
+
+In the following `GenericsExample` component:
+
+* The JS function `invokeDotNetMethodsAsync` is called when the **`Invoke Interop`** button is selected.
+* A pair of <xref:Microsoft.JSInterop.DotNetObjectReference> types are created and passed to the JS function for instances of the `GenericType` as a `string` and a `double`.
+
+`Pages/GenericsExample.razor`:
+
+```razor
+@page "/generics-example"
+@using System.Runtime.InteropServices
+@inject IJSRuntime JSRuntime
+
+<p>
+    <button @onclick="InvokeInterop">Invoke Interop</button>
+</p>
+
+<ul>
+    <li>genericType1: @genericType1?.Value</li>
+    <li>genericType2: @genericType2?.Value</li>
+</ul>
+
+@code {
+    private GenericType<string> genericType1 = new() { Value = "string 0" };
+    private GenericType<double> genericType2 = new() { Value = 0 };
```

Comment:
Can we do `string`/`int`? I don't see us really taking advantage of `double` in any way.

Reply:
Ah! Yes ... an oversight on my part because I went with an int on the JS side but didn't change it all the way through. I'm really, really tired and need a break 🏖️.